packages/mobile/src/components/challenge-rewards-drawer/ChallengeRewardsDrawerProvider.tsx

```diff
@@ -215,15 +215,17 @@ export const ChallengeRewardsDrawerProvider = () => {
           isCooldownChallenge={challenge && challenge.cooldown_days > 0}
           challengeState={challenge.state}
           currentStep={challenge.current_step_count}
-          stepCount={challenge.max_steps}
+          stepCount={challenge.max_steps ?? undefined}
```

Comment:
and then can you explain why this one is `undefined`?

Reply:
Just because the type in the props is `number | undefined` and I didn't want to dig into updating the prop shape in case that breaks something else.